desc/compute/_metric.py

```diff
@@ -2088,3 +2088,582 @@ def _cvdrift0(params, transforms, profiles, data, **kwargs):
+        )
+        + dot(data["e_rho|v,p"], cross(data["e_theta_PEST"], data["(e_phi_v)|PEST"]))
+    )
+    return data
+
+
+# TODO: Generalize for a general phi before #568
+@register_compute_fun(
+    name="(sqrt(g)_PEST_p)|PEST",
+    label="\\partial_{\\phi}|_{\\rho, \\vartheta} \\sqrt{g}_PEST",
+    units="m^{3}",
+    units_long="cubic meters",
+    description="Jacobian determinant of PEST coordinate system"
+    + " derivative w.r.t cylindrical toroidal angle",
+    dim=1,
+    params=[],
+    transforms={},
+    profiles=[],
+    coordinates="rtz",
+    data=[
+        "e_rho|v,p",
+        "e_theta_PEST",
+        "e_phi|r,v",
+        "(e_theta_PEST_p)|PEST",
+        "(e_rho_p)|PEST",
+        "(e_phi_p)|PEST",
+    ],
+)
+def _sqrtg_PEST_phi_PEST(params, transforms, profiles, data, **kwargs):
+    data["(sqrt(g)_PEST_p)|PEST"] = (
+        dot(data["(e_rho_p)|PEST"], cross(data["e_theta_PEST"], data["e_phi|r,v"]))
+        + dot(
+            data["e_rho|v,p"], cross(data["(e_theta_PEST_p)|PEST"], data["e_phi|r,v"])
+        )
+        + dot(data["e_rho|v,p"], cross(data["e_theta_PEST"], data["(e_phi_p)|PEST"]))
+    )
+    return data
+
+
+################################################################################
+#############--------------FINITE-n STABILITY METRICS--------------#############
+################################################################################
+
+
+@register_compute_fun(
+    name="B_dot_grad(grad(rho))",
+    label="\\nabla(\\nabla(\\rho))",
+    units=" T m^{-2}",
+    units_long="Tesla over square meters",
+    description="Gradient of gradient of rho tensor along the magnetic field",
+    dim=3,
+    params=[],
+    transforms={},
+    profiles=[],
+    coordinates="rtz",
+    data=[
+        "e_theta",
+        "e_zeta",
+        "e_theta_t",
+        "e_theta_z",
+        "e_zeta_t",
+        "e_zeta_z",
+        "sqrt(g)",
+        "sqrt(g)_t",
+        "sqrt(g)_z",
+        "B^theta",
+        "B^zeta",
+    ],
+)
+def _B_dot_grad_grad_rho(params, transforms, profiles, data, **kwargs):
+    # B⋅∇ = B^θ_P ∂/∂θ + B^ζ ∂/∂ζ
+    # ∇ρ = (e_θ × e_ζ)/√g
+
+    term1 = (
+        cross(data["e_theta_t"], data["e_zeta"]) / data["sqrt(g)"][:, jnp.newaxis]
+        + cross(data["e_theta"], data["e_zeta_t"]) / data["sqrt(g)"][:, jnp.newaxis]
+        - cross(data["e_theta"], data["e_zeta"])
+        * data["sqrt(g)_t"][:, jnp.newaxis]
+        / (data["sqrt(g)"][:, jnp.newaxis]) ** 2
+    ) * data["B^theta"][:, jnp.newaxis]
+
+    term2 = (
+        cross(data["e_theta_z"], data["e_zeta"]) / data["sqrt(g)"][:, jnp.newaxis]
+        + cross(data["e_theta"], data["e_zeta_z"]) / data["sqrt(g)"][:, jnp.newaxis]
+        - cross(data["e_theta"], data["e_zeta"])
+        * data["sqrt(g)_z"][:, jnp.newaxis]
+        / (data["sqrt(g)"][:, jnp.newaxis]) ** 2
+    ) * data["B^zeta"][:, jnp.newaxis]
+
+    data["B_dot_grad(grad(rho))"] = term1 + term2
```

Comment:
Please change all of this to just `data["(B*grad) grad(rho)"] = B^theta * e^rho_t + B^zeta * e^rho_z`. We already have those quantities registered. It will
1. Reduce computation
2. Reduce chance of math error
3. Make notation consistent with `(B*grad)B` quantity.